chunk.go

```diff
@@ -76,9 +76,17 @@ func (ch *Chunk) dump(w io.Writer, compressorIndex int) error {
 	return nil
 }
 
+type dummyCloser struct {
+	bytes.Buffer
+}
+
+func (c *dummyCloser) Close() error {
+	return nil
+}
+
 func compressData(src io.Reader, compressorIndex int) (*bytes.Buffer, error) {
-	compressed := new(bytes.Buffer)
```

Comment:
I think the function should be somehow like the following:

```go
func compressData(src io.Reader, compressorIndex int) (*bytes.Buffer, error) {
        var compressed bytes.Buffer
	var compressor io.WriteCloser

	switch compressorIndex {
	case NoCompression:
		compressor = &noopCompressor{compressed}
	case Snappy:
		compressor = snappy.NewBufferedWriter(compressed)
	case Gzip:
		compressor = gzip.NewWriter(compressed)
	default:
		return nil, fmt.Errorf("Unknown compression algorithm: %d", compressorIndex)
	}

        defer compressor.Close()
	if _, e := io.Copy(compressor, src); e != nil {
		return nil, fmt.Errorf("Failed to compress chunk data: %v", e)
	}

	return &compressed, nil
}
```

The point is that `compressed` is still of type `bytes.Buffer`, and we just create a `noopCompressor` which works like `snappy.Writer` and `gzip.Writer`.